src/main/kotlin/no/nav/syfo/kafka/Kafka.kt

```diff
@@ -21,10 +21,20 @@ suspend fun CoroutineScope.setupKafka(
     syketilfelleClient: SyketilfelleClient
 ) {
     val producerProperties = kafkaOversikthendelsetilfelleProducerProperties(env, vaultSecrets)
-    val oversikthendelseTilfelleProducer = KafkaProducer<String, KOversikthendelsetilfelle>(producerProperties)
+    val oversikthendelseTilfelleProducer: KafkaProducer<String, KOversikthendelsetilfelle>
 
     val producerOppfolgingstilfelleRetryProperties = kafkaOppfolgingstilfelleRetryProducerConfig(env, vaultSecrets)
-    val oppfolgingstilfelleRetryRecordProducer = KafkaProducer<String, KOppfolgingstilfelleRetry>(producerOppfolgingstilfelleRetryProperties)
+    val oppfolgingstilfelleRetryRecordProducer: KafkaProducer<String, KOppfolgingstilfelleRetry>
+
+    try {
+        oversikthendelseTilfelleProducer = KafkaProducer<String, KOversikthendelsetilfelle>(producerProperties)
+        oppfolgingstilfelleRetryRecordProducer = KafkaProducer<String, KOppfolgingstilfelleRetry>(producerOppfolgingstilfelleRetryProperties)
+    } catch (e: Exception) {
+        log.error("Fikk en feil ved oppretting av Kafka Producer, restarter pod", e)
+        state.running = false
+        throw e
+    }
+
```

Comment:
Hva med å heller flytte koden inn hver sin launchen slik at det dekkes at finally-blocken som allerede. Og i tillegg evt. legge på en en linje i de 2 finally-blockene:
`applicationState.initialized= false`

Reply:
Nå fikk vi exception før vi kom til `launch`, så det vil vel ikke fungere. `finally`-blockene har allerede `applicationState.running = false`, og det hjalp jo tydeligvis ikke.
Det vil vel heller ikke hjelpe å legge til `initialized`, da det brukes til "is_ready"-sjekken, og kubernetes restarter vel kun når "is_alive" feiler?

Reply:
`Nå fikk vi exception før vi kom til launch`. Ja fordi initalisering ikke ligger inne koden som kjører launch, 
I Isnarmesteleder så kjøres hele kafka-greia i launch i hvert fall og det ser ut som om det fungerer fint.

Reply:
Tror det er riktig at det er is_alive som må feile for å tvinge Kubernetes til å restarte pod'en. is_ready brukes av Kubernetes til å bestemme om trafikk kan rutes til pod'en (men det spiller ikke så stor rolle for denne app'en siden den ikke har noe api).